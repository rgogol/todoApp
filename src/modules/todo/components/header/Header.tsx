import React from 'react';
import { View, Text } from 'react-native';
import ShareButton from '../share';
import HeaderStyles from './styles';

const Header = () => {
  const { container, textStyles } = HeaderStyles;
  return (
    <View style={container}>
      <Text style={textStyles}>TODO APP</Text>
      <ShareButton />
    </View>
  );
};

export default Header;
