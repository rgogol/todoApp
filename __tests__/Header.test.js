import React from 'react';
import { shallow } from 'enzyme';
import Header from '../src/modules/todo/components/header';

jest.mock('react-native-share', () => {
  return {
    open: jest.fn(),
  };
});

describe('Header', () => {
  const wrapper = shallow(<Header />).dive();

  it('render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
