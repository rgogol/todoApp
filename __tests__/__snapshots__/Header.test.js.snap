// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#102e4a",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#cfd6ea",
        "fontSize": 18,
        "fontWeight": "700",
      }
    }
  >
    TODO APP
  </Text>
  <ShareButton />
</View>
`;
